README.md

```diff
@@ -143,6 +143,84 @@ All protobuf message classes are generated and available. Key classes include:
 
 See the `examples/` directory for complete working examples.
 
+## Authentication
+
+The SDK supports OAuth 2.0 Client Credentials flow for authentication with Kessel services. The `OAuth2ClientCredentials` class provides automatic token management with built-in refreshing.
+
+**Note**: To use authentication features, install the SDK with auth dependencies: `pip install "kessel-sdk[auth]"`
+
+### OAuth Configuration Options
+
+The SDK supports two ways to configure OAuth 2.0 authentication:
+
+#### Option 1: OIDC Discovery
+
+Use this approach when your OAuth provider supports OIDC discovery. The SDK provides a `fetchOIDCDiscovery` function to discover the token endpoint:
+
+```python
+import grpc
+import google.auth.transport.requests
+import google.auth.transport.grpc
+from kessel.grpc.auth import fetchOIDCDiscovery, OAuth2ClientCredentials
```

Comment:
`auth` should probably be outside of grpc package, maybe even at the same level i.e. `kessel.auth` instead.
This is something we could use for http as well

Reply:
Good point, makes sense according to the ddr and docs as well